File: #242 Valid Anagram/242.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t): return False
        ls = [0] * 26
        lt = [0] * 26
        for i in range(len(s)):
            ls[ord(s[i])-ord('a')] += 1
            lt[ord(t[i])-ord('a')] += 1
        for i in range(26):
            if ls[i] != lt[i]: return False
        return True

File: #242 Valid Anagram/242.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int slen = s.length(), tlen = t.length();
        if (slen!=tlen) return false;
        int ls[26+5]={0}, lt[26+5]={0};
        for(int i=0; i<slen; ++i)
            ls[s[i]-'a']++, lt[t[i]-'a']++;
        for(int i=0; i<26; ++i)
            if(ls[i]!=lt[i]) return false;
        return true;
    }
};